JUCX: Maven failed to install during jenkins test

http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9205/label=hpc-test-node3,worker=0/console
Looks like maven is using a shared folder (/labhome/swx-jenkins/.m2/repository) for jucx, which causes issues when multiple executors are running in parallel?
```
15:59:20 [INFO] Finished at: 2019-03-08T15:59:20+02:00
15:59:20 [INFO] Final Memory: 26M/733M
15:59:20 [INFO] ------------------------------------------------------------------------
15:59:20 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-install-plugin:2.4:install (default-install) on project jucx: Failed to install metadata org.ucx:jucx:0.0.1-SNAPSHOT/maven-metadata.xml: Could not read metadata /labhome/swx-jenkins/.m2/repository/org/ucx/jucx/0.0.1-SNAPSHOT/maven-metadata-local.xml: input contained no data -> [Help 1]
15:59:20 [ERROR] 
15:59:20 [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
15:59:20 [ERROR] Re-run Maven using the -X switch to enable full debug logging.
15:59:20 [ERROR] 
15:59:20 [ERROR] For more information about the errors and possible solutions, please read the following articles:
15:59:20 [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
15:59:20 make[4]: *** [install-data-hook] Error 1
15:59:20 make[4]: Leaving directory `/scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/0/build-test/bindings/java/src/main/native'
15:59:20 make[3]: *** [install-data-am] Error 2
15:59:20 make[3]: Leaving directory `/scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/0/build-test/bindings/java/src/main/native'
15:59:20 make[2]: *** [install-am] Error 2
15:59:20 make[2]: Leaving directory `/scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/0/build-test/bindings/java/src/main/native'
15:59:20 make[1]: *** [install] Error 2
15:59:20 make[1]: Leaving directory `/scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/0/build-test/bindings/java/src/main/native'
```

--- a/bindings/java/src/main/native/Makefile.am
+++ b/bindings/java/src/main/native/Makefile.am
@@ -5,7 +5,7 @@
 
 topdir=$(abs_top_builddir)
 javadir=$(top_srcdir)/bindings/java
-MVNCMD=$(MVN) -f $(javadir)/pom.xml
+MVNCMD=$(MVN) -B -f $(javadir)/pom.xml -Dmaven.repo.local=$(topdir)/.deps/
 
 BUILT_SOURCES = org_ucx_jucx_ucp_UcpConstants.h \
                 org_ucx_jucx_ucp_UcpContext.h \
@@ -57,3 +57,5 @@ clean-local:
 
 test:
 	$(MVNCMD) test
+docs:
+	$(MVNCMD) javadoc:javadoc

--- a/contrib/test_jenkins.sh
+++ b/contrib/test_jenkins.sh
@@ -224,9 +224,8 @@ build_java_docs() {
 	echo " ==== Building java docs ===="
 	if module_load dev/jdk && module_load dev/mvn
 	then
-		pushd ../bindings/java
-		mvn javadoc:javadoc
-		popd
+		../configure --prefix=$ucx_inst --with-java
+		$MAKE -C ../build-test/bindings/java/src/main/native docs
 		module unload dev/jdk
 		module unload dev/mvn
 	else
@@ -832,11 +831,9 @@ test_jucx() {
 	echo "1..2" > jucx_tests.tap
 	if module_load dev/jdk && module_load dev/mvn
 	then
-		pushd ../bindings/java/
 		export UCX_ERROR_SIGNALS=""
-		JUCX_INST=$ucx_inst mvn clean test
+		JUCX_INST=$ucx_inst $MAKE -C bindings/java/src/main/native test
 		unset UCX_ERROR_SIGNALS
-		popd
 		module unload dev/jdk
 		module unload dev/mvn
 		echo "ok 1 - jucx test" >> jucx_tests.tap

